browser/src/control/Toolbar.js

```diff
@@ -1085,7 +1085,7 @@ L.Map.include({
 			if ($('.leaflet-cursor').is(':visible'))
 				return;
 
-			if (window.mode.isMobile()) {
+			if (window.mode.isMobile() && this.isEditMode()) {
```

Comment:
what about readonly mode on mobile? it will try to show/hide some buttons in formulabar. is that correct?

Reply:
yes there are undo and redo buttons inside this scope  we need to hide it

Reply:
no, I mean "else" block below will be called in case of "readonly" or "non mobile"

Reply:
yes correct